lib/astutils.cpp

```diff
@@ -1050,6 +1130,8 @@ bool isOppositeCond(bool isNot, bool cpp, const Token * const cond1, const Token
             if (cond2->astOperand2() && cond2->astOperand2()->str() == "0")
                 return isSameExpression(cpp, true, cond1->astOperand1(), cond2->astOperand1(), library, pure, followVar, errors);
         }
+        if (!isUsedAsBool(cond2))
```

Comment:
Shouldn't this check both `cond1` and `cond2`?

Reply:
Here, `cond1` is `!`, so `isUsedAsBool(cond1)` is always true.

Reply:
Yes, but what if the parameters are flipped? We happen to pass `!` as cond1 in `OppositeExpressionAnalyser` but this is just arbitrary we could have passed `!` for cond2.

Reply:
Yes, this is already taken care of (I changed nothing there). If cond2 is `!`, isOppositeCond is calle dwith cond1 and cond2 swapped. ;-)

Reply:
I see, sorry, I missed that part.

Reply:
No worries.